modules/sirdataRtdProvider.js

```diff
@@ -0,0 +1,519 @@
+  var sendWithCredentials = true;
+
+  config.params.partnerId = config.params.partnerId ? config.params.partnerId : 1;
+  config.params.key = config.params.key ? config.params.key : 1;
+
+  if (userConsent.coppa) { // if children, no segments, page categories only
+    sirdataDomain = 'cookieless-data.com';
+    tcString = '';
+    sendWithCredentials = false;
+  } else if (userConsent.usp) {
+    // Si pas de transprence ou optout on passe en contextuel seulement
+    if (userConsent.usp[0] == '1' && (userConsent.usp[1] == 'N' || userConsent.usp[2] == 'Y')) {
+      sirdataDomain = 'cookieless-data.com';
+      sendWithCredentials = false;
+      gdprApplies = false;
+    }
+  } else if (gdprApplies && gdprApplies !== '0' && gdprApplies !== 'null') {
+    if (userConsent && userConsent.gdpr && userConsent.gdpr.vendorData && userConsent.gdpr.vendorData.vendor && userConsent.gdpr.vendorData.vendor.consents) {
+      if (!userConsent.gdpr.vendorData.vendor.consents[53] || !userConsent.gdpr.vendorData.purpose.consents[1] || !userConsent.gdpr.vendorData.purpose.consents[3]) {
+        sirdataDomain = 'cookieless-data.com';
+        sendWithCredentials = false;
+      }
+    } else {
+      sirdataDomain = 'cookieless-data.com';
+      sendWithCredentials = false;
+    }
+  }
+
+  var actualUrl = null;
+  try {
+    actualUrl = window.top.location.href;
+  } catch (e) {}
+  if (!actualUrl && config.params.actualUrl) {
+    try {
+      actualUrl = new URL(config.params.actualUrl);
+    } catch (err) {}
+  }
+
+  const url = 'https://kvt.' + sirdataDomain + '/api/v1/public/p/' + config.params.partnerId + '/d/' + config.params.key + '/s?callback=&gdpr=' + gdprApplies + '&gdpr_consent=' + tcString + (actualUrl ? '&url=' + actualUrl : '');
+  ajax(url, {
+    success: function (response, req) {
+      if (req.status === 200) {
+        try {
+          const data = JSON.parse(response);
+          if (data && data.segments) {
+            addSegmentData(adUnits, data, config, onDone, globalConfig);
+          } else {
+            onDone();
+          }
+        } catch (err) {
+          onDone();
+          utils.logError('unable to parse Sirdata data');
+        }
+      } else if (req.status === 204) {
+        onDone();
+      }
+    },
+    error: function () {
+      onDone();
+      utils.logError('unable to get Sirdata data');
+    }
+  },
+  null,
+  {
+    contentType: 'text/plain',
+    method: 'GET',
+    withCredentials: sendWithCredentials,
+    referrerPolicy: 'unsafe-url',
+    crossOrigin: true
+  });
+}
+
+export function setGlobalOrtb2(segments, categories, globalConfig) {
+  var ortb2Valid = true;
+
+  try {
+    if (parseFloat(globalConfig.version.substring(1)) < 4.3) {
+      ortb2Valid = false;
+    }
+  } catch (er) {}
+
+  if (!ortb2Valid) {
+    // do it with FPD
+    return setGlobalFpd(segments, categories, globalConfig);
+  }
+
+  try {
+    globalConfig.setConfig({
+      ortb2: {
+        site: {
```

Comment:
why not set site.content.data.ext also? What taxonomy are these segments coming from? If it is your own, do you publish the taxonomy? Are you interested in joining our discussion as #6057 matures?

Reply:
Yes we are interested in joining the discussion, and we mapped our taxonomy with Tech Lab's content taxonomy 2.2 already

Reply:
Please apply for community membership at prebid.org and contact @sfrancolla at Permutive for the meeting invite.